### PR TITLE
Add search messages method to Slack Connector

### PR DESCRIPTION
**Pull request by chillipeper**
# Description

Hello,
This PR allows opsdroid to search for messages in a channel, given a specific time range

## Status
**READY**

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Added relevant tests
-  using skills with the this new feature

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation (if applicable)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/connectors/slack.md [hunk 1]
```diff
@@ -87,6 +88,27 @@ class GreeterSkill(Skill):
         await message.respond("Hi")
 ```
 
+## Get messages from History
+Sometimes you need to search trough the history of a channel. For this you can use the `search_history_messages` method from the slack connector which returns all the messages on a specified range of time.
```
Comment (jacobtomlinson) [inline docs/connectors/slack.md:92]:
```suggestion
Sometimes you need to search through the history of a channel. For this you can use the `search_history_messages` method from the slack connector which returns all the messages on a specified range of time.
```

File: docs/connectors/slack.md [hunk 2]
```diff
@@ -87,6 +88,27 @@ class GreeterSkill(Skill):
...
+
+    @match_regex(r"search messages")
+    async def search_messages(self, message):
+        """"""
+        slack = self.opsdroid.get_connector("slack")
+        messages = await slack.search_history_messages(
+            "CHANEL_ID", start_time="1512085950.000216", end_time="1512104434.000490"
+        )
+        await message.respond(messages)
+```
```
Comment (jacobtomlinson) [inline docs/connectors/slack.md:109]: Could you move this into the docstring of the `search_history_messages` method and then use autofunc here to include it in the docs.

Helps avoid docs getting out of sync with code.

**Review by jacobtomlinson — COMMENTED** [review]
This sounds useful thanks! Couple of comments.

**Comment by chillipeper** [comment]
thanks for reviewing, will make sure to check it out this week

File: docs/connectors/slack.md (hunk 1 above)
Comment (chillipeper) [reply docs/connectors/slack.md]: Fixed

File: docs/connectors/slack.md (hunk 2 above)
Comment (chillipeper) [reply docs/connectors/slack.md]: Moved into the docstring

**Comment by chillipeper** [comment]
Maybe Read the Docs build is failing, because it's not using an up to date image?

```
WARNING: autodoc: failed to import function 'slack.ConnectorSlack.search_history_messages' from module 'opsdroid.connector'; the following exception was raised:
No module named 'slack_sdk'
```

**Review by jacobtomlinson — APPROVED** [review]

**Comment by jacobtomlinson** [comment]
Thanks @chillipeper

**Pull request merged**